### PR TITLE
Fixes 4670: show last error and status for templates

### PR DESCRIPTION
**Pull request by xbhouse**
## Summary

- Adds a status column to the template list page
- Displays any errors if there's an Invalid status

### Additional changes
- Also adds 2 commands to run the app against stage or prod (`npm run start:stage` and `npm run start:prod`) without needing to select from a menu - thank you @marusak for the suggestion!

## Testing steps

1. Add a custom and RH repository, let them snapshot, and create a template using latest content. The template status in the UI should change to In progress and then Valid once the update-template-content task completes.
2. Edit the custom repo URL used in the template to create a new snapshot. Once the snapshot finishes, stop the server and remove the pulp URL. Restart the server. The template status should now be Invalid and display the error from the update-latest-snapshot task when clicking on it.
3. Edit the template. The status should still be Invalid, with the error from the update-template-content task displayed on click.

For QE - it would be difficult to simulate an error here, so validating the happy path should be sufficient. I've added a check to see if we're in the ephemeral environment, so in ephemeral it will not depend on the task that only runs with candlepin enabled

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-4670

File: README.md [hunk 1]
```diff
@@ -36,7 +36,9 @@ Alternativly, simply add these lines to your /etc/hosts:
 
 4. `npm run build` (only required when setting up for the first time)
 
-5. `npm run start` to run against stage or prod environments. <br/>
+5. `npm run start:stage` to run against the stage environment <br/>
```
Comment (marusak) [inline README.md:39]: Thank you! However, shouldn't it be at least separate commit or maybe even PR? (Mostly interested here what is your git policy :) )

Comment (Andrewgdewar) [reply README.md:39]: I usually just add a separate section to my PR description for "Additional Changes" a separate commit is fine for this. 
I normally wouldn't bother with a separate PR, I find too tight of a commit policy results in me not fixing small bugs I otherwise would due to laziness.

Comment (xbhouse) [reply README.md:39]: thanks @Andrewgdewar, will clarify in the description and add in a separate commit for this!

File: src/Pages/Templates/TemplatesTable/TemplatesTable.tsx [hunk 2]
```diff
@@ -250,22 +280,26 @@ const TemplatesTable = () => {
...
                 </Tr>
               </Thead>
               <Tbody>
-                {templateList.map(
-                  (
-                    { uuid, name, description, arch, version, date, use_latest }: TemplateItem,
-                    index,
-                  ) => (
+                {templateList.map((rowData: TemplateItem, index) => {
+                  const { uuid, name, description, arch, version, date, use_latest } = rowData;
```
Comment (Andrewgdewar) [inline src/Pages/Templates/TemplatesTable/TemplatesTable.tsx:301]: #nit: I think you could keep the simplified direct return here. 
No need to destructure from the variable.

File: src/Pages/Templates/TemplatesTable/components/StatusIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+interface Props {
+  rowData: TemplateItem;
+}
+
+interface DescriptionProps {
+  error?: string;
+}
+
+const PopoverDescription = ({ error }: DescriptionProps) => <div>{error}</div>;
```
Comment (Andrewgdewar) [inline src/Pages/Templates/TemplatesTable/components/StatusIcon.tsx:26]: This component likely isn't needed.

**Review by Andrewgdewar — APPROVED** [review]
A couple of small #nit comments, everything else looks good.

**Review by Andrewgdewar — APPROVED** [review]
A couple of small #nit comments, everything else looks good.

File: src/Pages/Templates/TemplatesTable/TemplatesTable.tsx (hunk 2 above)
Comment (xbhouse) [reply src/Pages/Templates/TemplatesTable/TemplatesTable.tsx]: fix pushed

File: src/Pages/Templates/TemplatesTable/TemplatesTable.tsx [hunk 4]
```diff
@@ -326,7 +338,11 @@ const TemplatesTable = () => {
                         </ConditionalTooltip>
                       </Td>
                       <Td>
-                        <StatusIcon rowData={rowData} />
```
Comment (Andrewgdewar) [inline src/Pages/Templates/TemplatesTable/TemplatesTable.tsx]: oh....
I get it, apologies.

File: src/Pages/Templates/TemplatesTable/TemplatesTable.tsx (hunk 2 above)
Comment (Andrewgdewar) [reply src/Pages/Templates/TemplatesTable/TemplatesTable.tsx]: I see what you were doing there.. The change looks good, but what you had was likely fine. 😓

**Comment by swadeley** [comment]
Hi

If you create the template before the included RHEL repos, e.g. Ansible, are introspected the template shows Invalid and never becomes Valid after the RHEL repo reaches Valid status. Is that out if scope?

**Comment by xbhouse** [comment]
hi @swadeley, i think this is the same issue as the associated backend PR, because candlepin isn't in ephemeral and so the task that the status depends on doesn't run. i can adjust the logic here to check if we're in ephemeral so that the status isn't always Invalid, but we may only be able to test this fully in stage

**Comment by swadeley** [comment]
Hi

nothing broken in ephemeral, merging to test further in stage.
